AnimationProfiles/TPP/tpp_AP.pddl

```diff
@@ -0,0 +1,145 @@
+(define (animation TPP-Propositional)
+
+; Defines the Animation profile for the TPP propositional problem
+; 18/08/2023
+; By Shu Tong Luo
+;
+; Note: Given the nature of the problem domain, multiple depots / markets 
+; are visualised as overlapping as there is no initial state predicate.
+ 
+ 
+; Specifies the goods is loaded onto a truck at a given level
+   (:predicate loaded
+                  :parameters (?g ?t ?l)
+                  :effect(
+                  (equal (?t y) (add (?l y) 50))
+                  (equal (?g y) (add (?t y) 20))
+                  (assign (?g x) (function align_middle (objects ?g ?t)))
+                  )
+   )
+
+; Specifies the goods is stored at a given level
+   (:predicate stored
+                  :parameters (?g ?l)
+                  :effect (
+                    (equal (?g x) (add (?l x) 50))
+                    (assign (?g x y) (function distribute_within_objects_vertical (objects ?g ?l) (settings (spacebtw 30))))
+                  )
+   )
+
+; Specifies that goods are available at the market at a given level
+   (:predicate on-sale
+                  :parameters (?g ?m ?l)
+                  :effect (
+                  (equal (?g x) (add (?m x) 20))
+                  )
+   )
+
+; Specifies which level is on top of which level
+   (:predicate next
+                  :parameters (?l1 ?l2)
+                  :effect (
+                  (equal (?l1 x) (?l2 x))
+                  (equal (?l1 y) (add (?l2 y) 120))
+                  )
+   )
+
+; Specifies the position of the truck
+  (:predicate at
+                  :parameters (?t ?p)
+                  :effect (
+                  (assign (?t x) (function align_middle (objects ?t ?p)))
+                  (equal (?t y) (add (?p y) 50))
+                  )
+                  :priority 0
+  )
```

Comment:
Strangely, when the drive action is executed, the truck travels in a region without depots or markets. 
![image](https://github.com/planimation/documentation/assets/861782/8140c081-1415-4bc9-9570-ba3a96db355e)

We cannot see either the items loaded in the truck.